### PR TITLE
Fix bug: do not reuse axon branches and unravelled them

### PR DESCRIPTION
**Pull request by wizmer**
Axon branches were reused because the set() that keeps them
was being reset at every call of axon.repair().
Now this set is kept into the encompassing scope and passed through:
axon.repair(used_axon_branches)
    
Also, grafted axon branches are now unravelled before being used

### CODE REVIEW

**Comment by tomdele** [comment]
I have not seen anything wrong

File: neuror/main.py [hunk 1]
```diff
@@ -279,23 +281,27 @@ def __init__(self, inputfile, axons=None, seed=0, plane=None):
...
             self.neuron.write(outputfile)
             return
 
-        # The only purpose of the 'planes' variable is to keep
-        # each plane.morphology alive. Otherwise sections become unusable
+        # The only purpose of the 'keep_alive' variable is to keep
+        # morphologies alive. Otherwise sections become unusable
         # https://github.com/BlueBrain/MorphIO/issues/29
-        planes = []
+        keep_alive = []
```
Comment (mgeplf) [inline neuror/main.py:294]: Are the donor sections coming from a `mut` or 'normal' MorphIO object?  The above issues was for 'normal', and isn't a problem, afaik.  The `mut` version is this bug: https://github.com/BlueBrain/MorphIO/issues/161

**Comment by mgeplf** [comment]
Before merging this, can you post the pictures so we can look at them?

**Comment by wizmer** [comment]
Here is how it looks like before/after this PR:

Before:
axon-is-reused

After:
axon-not-reused

File: neuror/main.py (hunk 1 above)
Comment (wizmer) [reply neuror/main.py]: Yes, you're right. This is no longer needed. It indeed uses `morphio.Morphology` so we are good here.

**Comment by wizmer** [comment]
@mgeplf Can I merge this ?

**Comment by mgeplf** [comment]
If you're sure it's doing the right thing; I can't evaluate it at the moment

**Comment by wizmer** [comment]
I'm not sure but I think it's definitely better that previously. We can iterate later if we decide to make some changes.

**Comment by wizmer** [comment]
I would not mind some validation metrics from scientists though if possible.

**Pull request merged**